### ISSUE TITLE
Pyright parser uses wrong order of operations for binary operations

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#1681]
When binary operators are used in chains, pyright doesn't use the correct order of operations in some cases. See below for an example of a false positive error due to this issue.

```python
v1 = "1" in "1" == "1"  # Incorrectly generates error
v2 = ("1" in "1") == "1"
v3 = "1" in ("1" == "1") # Correctly generates error
```

**Comment by erictraut** [comment]
This turned out not to be a parsing issue but rather a case where "comparison chaining" was not being applied correctly. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is included in pyright 1.1.125, which I just published. It will also be included in the next release of pylance.
